### DISCUSSION TITLE
Best way to trigger email when create custome function user registration

### DISCUSSION THREAD

**Discussion by dvrg** [pallets-eco/flask-security#480] [Q&A] [answered]
Hi, thanks for the work for this amazing packages

I going to make my own apps thats make user registrations with custome functions and views (registration for customer pov), it's very easy to create because `user_datastore` function, but how i can trigger emails easy way instead create extended MyUtil.send_mail function ? 

```python
@bp.route(f"/{PREFIX}/create", methods=["GET", "POST"])
@auth_required("session")
@roles_required("Admin")
def user_create():
    form = ExtendedRegisterForm()
    if form.validate_on_submit():
        """ create new user """
        new_user = user_datastore.create_user(
            email=form.email.data, password=hash_password(form.password.data)
        )
        if form.role.data:
            """ if role form exist, so add role to user """
            user_datastore.add_role_to_user(new_user, form.role.data)
        db.session.commit()
        # trigger the email
        flash(_default_messages["SUCCESS"][0], _default_messages["SUCCESS"][1])
        return redirect(url_for("back.user"))
    return render_template(
        bp.__getattribute__("name") + f"/{PREFIX}/" + "create.html",
        page_title=PREFIX,
        register_user_form=form,
    )
````

Thank you. I hope any body can give me suggest.

**Comment by jwag956** [comment, marked as answer]
You don't need to subclass MailUtil - I presume you have already configured an email program - such as Flask-Mail - and that is working. Where you want to send an email - just call the appropriate flask_mail methods - e.g. send_mail. You can look at the source for flask_security mail_util.py to see the 5 or so lines of code necessary.

**Reply by dvrg** [reply]
okey, thank you @jwag956 for suggestion, i will try it!
